tests/Aspire.Dashboard.Tests/Integration/Playwright/PlaywrightTestsBase.cs

```diff
@@ -50,4 +62,24 @@ public async ValueTask DisposeAsync()
             await _context.DisposeAsync();
         }
     }
+
+    private async Task TakeScreenshotAsync(string path, IPage page)
+    {
+        var screenshotPath = Path.Combine(GetLogPath(), path);
+        await page.ScreenshotAsync(new PageScreenshotOptions { Path = screenshotPath });
+        _output.WriteLine($"Dashboard screenshot saved to {screenshotPath}");
+    }
+
+    private string GetLogPath()
+    {
+        var testLogPath = Path.Combine("artifacts", "log");
```

Comment:
This will create a `artifacts/log` path under your `bindir`.
https://github.com/dotnet/aspire/blob/4938cb30aa8f7a9f535cc8a5d3735a79a4f8e0d3/tests/Aspire.Hosting.Tests/MSBuildTests.cs#L126-L137

This method can be used to find the repo root, and then `+ "/artifacts/log/"` should get you the correct path. It might be useful to move this method somewhere usable by other tests too.

Also, try inducing a failure and check that you can retrieve the screenshot from the logs.